internal/status/status.go

```diff
@@ -60,3 +61,9 @@ func (DefaultStatusHandler) SetDeleting(ctx context.Context, client client.Clien
 	istioCR.Status.State = operatorv1alpha1.Deleting
 	return update(ctx, client, istioCR, condition, retryTime...)
 }
+
+func (DefaultStatusHandler) SetWarning(ctx context.Context, err described_errors.DescribedError, client client.Client, istioCR *operatorv1alpha1.Istio, condition metav1.Condition, retryTime ...time.Duration) (ctrl.Result, error) {
```

Comment:
As warning is also a `described_error` I think that moving the logic under `SetError` func would make more sense (so we should set `istioCR.Status.State` based on err.level()). Then the new logic in https://github.com/kyma-project/istio/blob/fa38c94d0718c4153f1d2adf55452206d3b0edba/controllers/istio_controller.go#L84 could be removed, only leaving SetError at the end.